base/int.jl

```diff
@@ -305,6 +305,9 @@ promote_rule{T<:BitSigned64}(::Type{UInt64}, ::Type{T}) = UInt64
 promote_rule{T<:Union{UInt32, UInt64}}(::Type{T}, ::Type{Int128}) = Int128
 promote_rule{T<:BitSigned}(::Type{UInt128}, ::Type{T}) = UInt128
 
+# the result on one is a good heuristic for promotion type
+promote_op{T,R<:Integer}(::Type{T}, ::Type{R}) = T  # to avoid ambiguity
+promote_op{R<:Integer}(op, ::Type{R}) = typeof(op(one(R)))
```

Comment:
That sounds like a possible partial workaround to https://github.com/JuliaLang/julia/issues/16164 until we get a more general solution, but I think we should discuss this choice on that issue, and keep this PR focused on nullables. Please remove this commit for now to increase the changes of merging the PR soon.